week02/LongestCommonPrefix.cpp

```diff
@@ -0,0 +1,35 @@
+//TC: size of all strings
+//MC: O(1)
+class Solution {
+public:
+    string longestCommonPrefix(vector<string>& strs) {
+        int sz = strs.size();
+
+        if(!sz) return "";
+
+        int longestPrefLen = INT_MAX;
+
+        for(int i=0; i<sz; ++i){
+	        longestPrefLen = min(longestPrefLen, getCommonPrefLength(strs[0], strs[i]));
+        }
+
+        string longestPref = "";
+        for(int i=0; i<longestPrefLen; ++i){
+	        longestPref+= strs[0][i];
+        }
+
+        return longestPref;
```

Comment:
```suggestion
        return strs[0].substr(0, longestPrefLen);
```

You could just create a substring directly from `strs[0]` of length `longestPrefLen`. The code would be much cleaner.